test/unit/test-amp-story-player.js

```diff
@@ -691,58 +689,93 @@ describes.realWin('AmpStoryPlayer', {amp: false}, (env) => {
       return expect(() => player.go(-1)).to.throw('Out of Story range.');
     });
 
-    it('signals when player changed story using next method', async () => {
+    it('takes to first story when swiping on the last one with circular wrapping', async () => {
       const playerEl = win.document.createElement('amp-story-player');
       appendStoriesToPlayer(playerEl, 5);
 
       const player = new AmpStoryPlayer(win, playerEl);
 
       await player.load();
+      await nextTick();
 
       const navigationSpy = env.sandbox.spy();
       playerEl.addEventListener('navigation', navigationSpy);
-      player.next_();
-      expect(navigationSpy.firstCall.args[0].type).to.eql('navigation');
-      expect(navigationSpy.firstCall.args[0].detail).to.eql({
-        index: 1,
-        remaining: 3,
+      playerEl.setAttribute('enable-circular-wrapping', '');
+      player.go(4);
+      swipeLeft();
```

Comment:
I think this won't work because the messaging stub is only being created in the `buildStoryPlayer()` function inside the tests. And since you don't call that specific function in this test, the messaging stub is not available.

Can you move this to be the last thing in the `beforeEach()` function? That will make it so that any test can use the fake messaging:

https://github.com/ampproject/amphtml/blob/dd1457897303f45915aeaee3b3bc186954aabbbd/test/unit/test-amp-story-player.js#L66-L68

cc @connielei who also ran into this problem